pkg/R/sparkR.R

```diff
@@ -110,9 +110,19 @@ sparkR.init <- function(
                 mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
                 args = as.character(sparkRBackendPort),
                 javaOpts = paste("-Xmx", sparkMem, sep = ""))
-  Sys.sleep(2) # Wait for backend to come up
+
+  cat("Waiting JVM bring up ...\n")
+  while(TRUE) {
+    if(!connExists(.sparkREnv)) {
```

Comment:
Yeah, so I think the contract we are assuming here is that if `connectBackend` succeeds then it sets a valid `.sparkRCon` in `.sparkREnv`. So I think thats fine to assume that. 

I think @felixcheung 's suggestion of moving line 125 (old line 114) above this block would be good to do even though it doesn't affect correctness here.